File: test_modules/guest_modules/java/src/guest/Interfaces.java
package guest;

public class Interfaces {
	public interface Greeter {
		String greet(String name);
	}

	public static class SimpleGreeter implements Greeter {
		@Override
		public String greet(String name) {
			return "Hello " + name;
		}
	}

	public static String callGreeter(Greeter greeter, String name) {
		return greeter.greet(name);
	}
}

File: test_modules/guest_modules/go/interfaces.go
package guest

type Greeter interface {
	Greet(name string) string
}

type SimpleGreeter struct{}

func (g SimpleGreeter) Greet(name string) string {
	return "Hello " + name
}

func CallGreeter(g Greeter, name string) string {
	return g.Greet(name)
}
